File: matlab/RGB.m
function img = initImageRGB(h, l)
    img = randi([0 255], 3, h, l);
end

function imgH = symetrieH(img)
    imgH = img(:, end:-1:1, :);
end

function imgV = symetrieV(img)
    imgV = img(:, :, end:-1:1);
end

function imgray = grayscale(imageRGB)
    [~, n, m] = size(imageRGB);
    imgray = zeros(n, m);

    for i = 1:n
        for j = 1:m
            r = imageRGB(1, i, j);
            g = imageRGB(2, i, j);
            b = imageRGB(3, i, j);

            maxi = max([r g b]);
            mini = min([r g b]);

            imgray(i, j) = floor((maxi + mini) / 2);
        end
    end
end

File: python/RGB.py
import numpy as np
import matplotlib.pyplot as plt
from random import randrange

def	initImageRGB(imageRGB):
    for c in range (3):
        for i in range (imageRGB[0]):
            for j in range (imageRGB[0][0]):
                imageRGB[c][i][j] = randrange(256)
    return imageRGB

def	 symetrieH(img):
    return img[:, ::-1, :]

def	 symetrieV(img):
    return img[:, :, ::-1]

def	grayscale(imageRGB):
    _, n, m = imageRGB.shape
    imgray = np.zeros(n, m, dtype = int)
    for i in range (n):
        for j in range (m):
            r = imageRGB[0][i][j]
            g = imageRGB[1][i][j]
            b = imageRGB[2][i][j]
            maxi = max(r, g, b)
            mini = min(r, g, b)
            
            imgray[i][j] = (maxi + mini) // 2
        
    return imgray

